### PR TITLE
Csp 785 get calendar events

### PR DESCRIPTION
**Pull request by MarkFCain**

### CODE REVIEW

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/CalendarEvents/Queries/GetCalendarEvents/CalendarEvent.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    public string? EventFormat { get; set; }
+    public DateTime Start { get; set; }
+    public DateTime End { get; set; }
+
+    public string? Description { get; set; }
+    public string? Summary { get; set; }
+    public string? Location { get; set; }
+    public string? Postcode { get; set; }
+    public double? Longitude { get; set; }
+    public double? Latitude { get; set; }
```
Comment (DevenKShah) [inline src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/CalendarEvents/Queries/GetCalendarEvents/CalendarEvent.cs:15]: Make sure only nullables are set to nullable, to by pass the warning we can use `=null!;`

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application.UnitTests/CalendarEvents/Queries/GetCalendarEvents/GetCalendarEventsQueryTests.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+﻿using FluentAssertions;
+using SFA.DAS.ApprenticeAan.Application.CalendarEvents.Queries.GetCalendarEvents;
+using SFA.DAS.Testing.AutoFixture;
+
+namespace SFA.DAS.ApprenticeAan.Application.UnitTests.CalendarEvents.Queries.GetCalendarEvents;
+public class GetCalendarEventsQueryTests
+{
+    [Test, RecursiveMoqAutoData]
```
Comment (DevenKShah) [inline src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application.UnitTests/CalendarEvents/Queries/GetCalendarEvents/GetCalendarEventsQueryTests.cs:8]: Only AutoData will do here

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application.UnitTests/CalendarEvents/Queries/GetCalendarEvents/GetCalendarEventsQueryHandlerTests.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+using SFA.DAS.ApprenticeAan.Application.CalendarEvents.Queries.GetCalendarEvents;
+using SFA.DAS.ApprenticeAan.Application.Infrastructure;
+using SFA.DAS.Testing.AutoFixture;
+
+namespace SFA.DAS.ApprenticeAan.Application.UnitTests.CalendarEvents.Queries.GetCalendarEvents;
+public class GetCalendarEventsQueryHandlerTests
+{
+    [Test]
+    [MoqAutoData]
+    public async Task Handle_ReturnAllRegions(
```
Comment (DevenKShah) [inline src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application.UnitTests/CalendarEvents/Queries/GetCalendarEvents/GetCalendarEventsQueryHandlerTests.cs:13]: method name is wrong

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Controllers/CalendarEventsController.cs [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    }
+
+    [HttpGet]
+    [Route("")]
+    [ProducesResponseType(typeof(GetCalendarEventsQueryResult), StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<IActionResult> GetCalendarEvents([FromHeader(Name = Constants.ApiHeaders.RequestedByMemberIdHeader)] Guid requestedByMemberId, CancellationToken cancellationToken)
+    {
+        var response = await _mediator.Send(new GetCalendarEventsQuery(requestedByMemberId), cancellationToken);
+        if (response == null)
```
Comment (DevenKShah) [inline src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Controllers/CalendarEventsController.cs:26]: NotFound should not be the case

**Review by DevenKShah — COMMENTED** [review]
My comments reflects some of the changes that I have requested in the inner api

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Controllers/CalendarEventsController.cs [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    public CalendarEventsController(IMediator mediator)
+    {
+        _mediator = mediator;
+    }
+
+    [HttpGet]
+    [Route("")]
+    [ProducesResponseType(typeof(GetCalendarEventsQueryResult), StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<IActionResult> GetCalendarEvents([FromHeader(Name = Constants.ApiHeaders.RequestedByMemberIdHeader)] Guid requestedByMemberId, CancellationToken cancellationToken)
```
[the remark this answers is not in the document]
Comment (DevenKShah) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Controllers/CalendarEventsController.cs:23]: Simple logging can be done by using filters. Don't want to clutter methods just to log incoming requests.

[the remark this answers is not in the document]
Comment (DevenKShah) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Controllers/CalendarEventsController.cs:23]: BTW Application Insight captures all the incoming request and responses. So unless there is a specific need to do any additional logging, we can avoid it.

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application.UnitTests/CalendarEvents/Queries/GetCalendarEvents/GetCalendarEventsQueryRequestTests.cs [hunk 6]
```diff
@@ -0,0 +1,14 @@
+﻿using FluentAssertions;
+using SFA.DAS.ApprenticeAan.Application.InnerApi.CalendarEvents;
+using SFA.DAS.Testing.AutoFixture;
+
+namespace SFA.DAS.ApprenticeAan.Application.UnitTests.CalendarEvents.Queries.GetCalendarEvents;
+public class GetCalendarEventsQueryRequestTests
+{
+    [Test, RecursiveMoqAutoData]
+    public void Request_CheckGetUrl()
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application.UnitTests/CalendarEvents/Queries/GetCalendarEvents/GetCalendarEventsQueryRequestTests.cs:9]: file redundant, removed

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application.UnitTests/CalendarEvents/Queries/GetCalendarEvents/GetCalendarEventsQueryTests.cs (hunk 2 above)
Comment (MarkFCain) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application.UnitTests/CalendarEvents/Queries/GetCalendarEvents/GetCalendarEventsQueryTests.cs:8]: ok

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/CalendarEvents/Queries/GetCalendarEvents/CalendarEvent.cs (hunk 1 above)
Comment (MarkFCain) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/CalendarEvents/Queries/GetCalendarEvents/CalendarEvent.cs:15]: will use that

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/InnerApi/CalendarEvents/GetCalendarEventsQueryRequest.cs
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/InnerApi/CalendarEvents/GetCalendarEventsQueryRequest.cs:1]: removed

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Controllers/CalendarEventsController.cs (hunk 4 above)
Comment (MarkFCain) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Controllers/CalendarEventsController.cs:26]: ok, removed

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/CalendarEvents/Queries/GetCalendarEvents/GetCalendarEventsQueryHandler.cs [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+        _apiClient = apiClient;
+    }
+
+    public async Task<GetCalendarEventsQueryResult?> Handle(GetCalendarEventsQuery request,
+        CancellationToken cancellationToken)
+    {
+
+        var response = await _apiClient.GetCalendarEvents(request.RequestedByMemberId.ToString(), cancellationToken);
+
+        if (response.ResponseMessage.StatusCode == HttpStatusCode.OK)
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/CalendarEvents/Queries/GetCalendarEvents/GetCalendarEventsQueryHandler.cs:21]: nice!

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Controllers/CalendarEventsController.cs [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+{
+    private readonly IMediator _mediator;
+
+    public CalendarEventsController(IMediator mediator)
+    {
+        _mediator = mediator;
+    }
+
+    [HttpGet]
+    [Route("")]
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Controllers/CalendarEventsController.cs:20]: no, removed now

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application.UnitTests/CalendarEvents/Queries/GetCalendarEvents/GetCalendarEventsQueryTests.cs [hunk 9]
```diff
@@ -0,0 +1,14 @@
+﻿using AutoFixture.NUnit3;
+using FluentAssertions;
+using SFA.DAS.ApprenticeAan.Application.CalendarEvents.Queries.GetCalendarEvents;
+
+namespace SFA.DAS.ApprenticeAan.Application.UnitTests.CalendarEvents.Queries.GetCalendarEvents;
+public class GetCalendarEventsQueryTests
+{
+    [Test, AutoData]
+    public void Operator_PopulatesModelFroParameter(Guid memberId)
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application.UnitTests/CalendarEvents/Queries/GetCalendarEvents/GetCalendarEventsQueryTests.cs:9]: ok

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/Infrastructure/IAanHubRestApiClient.cs [hunk 10]
```diff
@@ -16,11 +17,7 @@ public interface IAanHubRestApiClient
...
 
-    /// [Get("calendarEvents")]
-    /// [AllowAnyStatusCode]
-    /// Task<Response<GetCalendarEventsQueryResult>> GetCalendarEvents(
-    ///     [Header(ApiHeaders.RequestedByMemberId)] string requestedByMemberId);
-
-
-
+    [Get("calendarEvents")]
+    [AllowAnyStatusCode]
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/Infrastructure/IAanHubRestApiClient.cs:21]: removed now.  Works fine when smoke tested, nice change

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api.UnitTests/Controllers/CalendarEventsControllerGetCalendarEventsTests.cs [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+    {
+        await sut.GetCalendarEvents(requestedByMemberId, cancellationToken);
+
+        mediatorMock.Verify(
+            m => m.Send(It.Is<GetCalendarEventsQuery>(q => q.RequestedByMemberId == requestedByMemberId),
+                It.IsAny<CancellationToken>()));
+    }
+
+    [Test]
+    [RecursiveMoqAutoData]
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api.UnitTests/Controllers/CalendarEventsControllerGetCalendarEventsTests.cs:29]: ok

**Review by DevenKShah — APPROVED** [review]

**Pull request merged**